src/layouts/app/translations-mixin.ts

```diff
@@ -92,9 +96,12 @@ export default (superClass: Constructor<LitElement & HassBaseEl>) =>
     private _selectLanguage(event) {
       const language: string = event.detail.language;
       this._updateHass({ language, selectedLanguage: language });
+      if (event.detail.save) {
+        storeState(this.hass!);
+        saveFrontendUserData(this.hass!, "language", language);
```

Comment:
Let's not just store a string, but instead store an object. It will allow us to store other language specific things in the future too. 

```suggestion
        saveFrontendUserData(this.hass!, "language", { language });
```